### PR TITLE
[AzureMonitorDistro] Add LoggerProviderBuilder extensions and use WithTracing in distro.

### PR DESCRIPTION
**Pull request by rajkumar-rangaraj**
- Add `LoggerProviderBuilder` extensions to Azure Monitor Exporter.
- Modified the internal logic in distro to migrate from `builder.Services.AddLogging` to `builder.WithLogging`

### CODE REVIEW

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/CHANGELOG.md [hunk 1]
```diff
@@ -2,6 +2,11 @@
 
 ## 1.4.0-beta.1 (2024-06-11)
 
+### Features Added
+
+* Added `LoggerProviderBuilder.AddAzureMonitorLogExporter` registration extensions.
```
Comment (vishweshbankwar) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/CHANGELOG.md:7]:
```suggestion
* Added `LoggerProviderBuilder.AddAzureMonitorLogExporter` registration extension.
```

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorExporterExtensions.cs [hunk 2]
```diff
@@ -59,6 +59,7 @@ public static class AzureMonitorExporterExtensions
                 throw new InvalidOperationException("The provided TracerProviderBuilder does not implement IDeferredTracerProviderBuilder.");
             }
 
+            // TODO: Do we need provide an option to alter BatchExportActivityProcessorOptions?
```
Comment (vishweshbankwar) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorExporterExtensions.cs:62]: Same todo for BatchExportLogRecordProcessorOptions.

File: sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorExporterExtensions.cs [hunk 3]
```diff
@@ -180,5 +181,67 @@ public static class AzureMonitorExporterExtensions
...
+                }
+
+                if (credential != null)
+                {
+                    // Credential can be set by either AzureMonitorExporterOptions or Extension Method Parameter.
+                    // Options should take precedence.
+                    exporterOptions.Credential ??= credential;
+                }
+
+                // TODO: Do we need provide an option to alter BatchExportActivityProcessorOptions?
```
Comment (vishweshbankwar) [inline sdk/monitor/Azure.Monitor.OpenTelemetry.Exporter/src/AzureMonitorExporterExtensions.cs:242]:
```suggestion
                // TODO: Do we need provide an option to alter BatchExportLogRecordProcessorOptions?
```

**Review by vishweshbankwar — APPROVED** [review]

**Review by TimothyMothra — APPROVED** [review]
LGTM!

**Pull request merged**